src/library_wasmfs.js

```diff
@@ -86,12 +86,17 @@ mergeInto(LibraryManager.library, {
 
       // Copy the file into a JS buffer on the heap.
       var buf = __wasmfs_read_file(pathName);
-      // The integer length is returned in the first 8 bytes of the buffer.
+      // The integer length resides in the first 8 bytes of the buffer.
       var length = {{{ makeGetValue('buf', '0', 'i64') }}};
 
       // Default return type is binary.
       // The buffer contents exist 8 bytes after the returned pointer.
+#if WASM_BIGINT
+      // The length as BigInt must be converted to a Number.
+      var ret = new Uint8Array(HEAPU8.subarray(buf + 8, buf + 8 + Number(length)));
```

Comment:
I was considering using `$bigintToI53Checked` here instead of `Number(length)`.
https://github.com/emscripten-core/emscripten/blob/5003f946206322854a9ee4adfcde01d3b323e5a4/src/library_int53.js#L127-L136
That is, something similar to `{{{ receiveI64ParamAsI53('length', -cDefine('EOVERFLOW')) }}}`.
https://github.com/emscripten-core/emscripten/blob/5003f946206322854a9ee4adfcde01d3b323e5a4/src/parseTools.js#L1118-L1122

But I think this is redundant, since that is only a issue when using >= ~8GB buffers (`_malloc(Number.MAX_SAFE_INTEGER + 1)`). I'm not aware of any VM that currently supports such large buffers.

Reply:
Agreed, that size buffer should not be a problem.

But actually I think that using `readI53FromU64` may be the best thing here. We do actually want a JS number here for the reason you said, and that seems like the simplest way to get it. Then we wouldn't need an ifdef here: `readI53FromU64` returns a Number either way.

@sbc100 what do you think?

Reply:
Perhaps we should modify `makeGetValue` above such that we can pass `i53` instead of `i64` ?     This would differ from `i64` in that the return value would always be a Number?

Reply:
Oh, there is a difference between the signed and unsigned case though, which is why we have both `readI53FromU64` and `readI53FromI64`, so just `i53` is not enough information I think.

Reply:
We already support i64 and u64 in makeGetValue so we can support i53 and u53 too.

Reply:
Oh, I forgot that, i53/u53 might make sense then.

One issue is that functions need to include the helper function as a dep. But maybe the auto-dep stuff you recently added could make that work automatically.

Reply:
We have `i53ConversionDeps` for this purpose.  Can makeGetValue not just call `bigintToI53Checked`... I don't see any singed/unsigned version of `bigintToI53Checked`..

Reply:
Commit 8efae3e0ef2bd5899e844f90da3e7ded913a87b1 incorporates this (but this will probably conflict with PR #17459). According to POSIX, `off_t` is a signed integer type, so `readI53FromU64` is not appropriate here.

Note that this will not catch overflows, see e.g. these TODO items:
https://github.com/emscripten-core/emscripten/blob/a41b1a37d25f85c85c269150f49172642a93b73d/src/library_int53.js#L84
https://github.com/emscripten-core/emscripten/blob/a41b1a37d25f85c85c269150f49172642a93b73d/src/library_int53.js#L91